### PR TITLE
Python: `warpx.multifab` legacy signature

### PR DESCRIPTION
**Pull request by ax3l**
Keep the legacy signature of the fully qualified name a bit longer to avoid breakage. Throw warnings for users to migrate.

Follow-up to #5230

### CODE REVIEW

File: Docs/source/usage/workflows/python_extend.rst [hunk 1]
```diff
@@ -134,9 +134,12 @@ This example accesses the :math:`E_x(x,y,z)` field at level 0 after every time s
        warpx = sim.extension.warpx
 
        # data access
-       E_x_mf = warpx.multifab(f"Efield_fp[x][level=0]")
+       #   vector field E, component x, on the fine patch of MR level 0
+       E_x_mf = warpx.multifab(f"Efield_fp", dir=0, level=0)
```
Comment (dpgrote) [inline Docs/source/usage/workflows/python_extend.rst]: A quick comment - the `f` prefix is not needed on the strings.

Comment (ax3l) [reply Docs/source/usage/workflows/python_extend.rst]: True! not sure how this slipped in, maybe a more general example.

**Review by RemiLehe — APPROVED** [review]

File: Source/Python/WarpX.cpp [hunk 2]
```diff
@@ -120,17 +120,27 @@ void init_WarpX (py::module& m)
...
-                if (wx.m_fields.has(multifab_name, level)) {
-                    return wx.m_fields.get(multifab_name, level);
+             [](WarpX & wx, std::string internal_name) {
+                 if (wx.m_fields.internal_has(internal_name)) {
+                     return wx.m_fields.internal_get(internal_name);
+                 } else {
+                     throw std::runtime_error("MultiFab '" + internal_name + "' is unknown or is not allocated!");
+                 }
+             },
+             py::arg("internal_name")
```
Comment (ax3l) [inline Source/Python/WarpX.cpp:130]: Forgot a
```suggestion
             py::arg("internal_name"),
             py::return_value_policy::reference_internal
```

Comment (ax3l) [reply Source/Python/WarpX.cpp:130]: Fix in #5325

**Pull request merged**